components/dashboard/src/utils.ts

```diff
@@ -213,3 +213,31 @@ export class ReplayableEventEmitter<EventTypes extends EventMap> extends EventEm
         return this.reachedEnd;
     }
 }
+
+function isValidURL(url: string) {
+    try {
+        new URL(url);
+        return true;
+    } catch (_) {
+        return false;
+    }
+}
+
+export function isTrustedUrlOrPath(urlOrPath: string) {
```

Comment:
We're essentially now constructing the URL twice. What if we mimicked the `URL.parse`? We'd return the URL if it's valid and `null` if it's not. If it's a URL, we can do our further checking, and if not, we return the prefix-based matching result, getting rid of the entire `try...catch` within the function.

Reply:
@filiptronicek do you mean to have function like `function parseTrustedUrlOrPath(urlOrPath: string): string` instead?

Reply:
@mustard-mh nope, I would have a `parseUrl` instead of a `isValidUrl`.

Reply:
`isTrustedUrlOrPath` should keep returning a boolean

Reply:
Addressed 👍 
<img width="1044" alt="image" src="https://github.com/user-attachments/assets/b6ce529e-3753-4707-b81f-2fe07485621d">